babel/src/main/java/org/apache/calcite/sql/babel/SqlCreateFunctionSqlForm.java

```diff
@@ -0,0 +1,171 @@
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.calcite.sql.babel;
+
+import org.apache.calcite.sql.SqlCreate;
+import org.apache.calcite.sql.SqlDataTypeSpec;
+import org.apache.calcite.sql.SqlIdentifier;
+import org.apache.calcite.sql.SqlKind;
+import org.apache.calcite.sql.SqlNode;
+import org.apache.calcite.sql.SqlSpecialOperator;
+import org.apache.calcite.sql.SqlWriter;
+import org.apache.calcite.sql.parser.SqlParserPos;
+import org.apache.calcite.util.ImmutableNullableList;
+
+import java.util.List;
+
+public class SqlCreateFunctionSqlForm extends SqlCreate {
+  public final SqlIdentifier functionName;
+  public final SqlIdentifier specificFunctionName;
+  public List<SqlIdentifier> fieldNames;
+  public List<SqlDataTypeSpec> fieldTypes;
+  public final SqlDataTypeSpec returnsDataType;
+  public final DeterministicType isDeterministic;
+  public final ReactToNullInputType canRunOnNullInput;
+  public final boolean hasSqlSecurityDefiner;
+  public final int typeInt;
+  public final SqlNode returnExpression;
+
+  private static final SqlSpecialOperator OPERATOR =
+      new SqlSpecialOperator("CREATE FUNCTION", SqlKind.CREATE_FUNCTION);
+
+  /** Creates a SqlCreateFunctionSqlForm.
+   * @param pos position
+   * @param replace if "or replace" token occurred
+   * @param functionName the name of the function
+   * @param specificFunctionName an optional specific functionName
+   * @param fieldNames function parameter names
+   * @param fieldTypes function parameter types
+   * @param returnsDataType return type of the function
+   * @param isDeterministic if "deterministic" is specified
+   * @param canRunOnNullInput if "called on null input" or "returns null on null input"
+   *                          is specified
+   * @param hasSqlSecurityDefiner if "sql security definer" is specified
+   * @param typeInt integer value after inline type
+   * @param returnExpression the expression that is returned
+   * */
+  public SqlCreateFunctionSqlForm(final SqlParserPos pos, final boolean replace,
+      final SqlIdentifier functionName,
+      final SqlIdentifier specificFunctionName, final List<SqlIdentifier> fieldNames,
+      final List<SqlDataTypeSpec> fieldTypes,
+      final SqlDataTypeSpec returnsDataType, final DeterministicType isDeterministic,
+      final ReactToNullInputType canRunOnNullInput, final boolean hasSqlSecurityDefiner,
+      final int typeInt, final SqlNode returnExpression) {
+
```

Comment:
nit: extra new line

Reply:
I keep this new line here to improve the readability of the code. The constructor takes in a large amount of argument, therefore it would look like a cluster with the code in the constructor without this new line. I am leaning toward keeping it. Please let me know your opinion on it.

Reply:
Yeah I see what you mean. I would probably keep it in that case